diff --git a/tests/integration/pilot/vm_test.go b/tests/integration/pilot/vm_test.go
index 5aa5af8855..8563d7fdb4 100644
--- a/tests/integration/pilot/vm_test.go
+++ b/tests/integration/pilot/vm_test.go
@@ -100,7 +100,7 @@ func TestVMRegistrationLifecycle(t *testing.T) {
 						return err
 					}
 					return res.CheckOK()
-				}, retry.Timeout(15*time.Second))
+				}, retry.Timeout(5*time.Minute))
 			})
 			ctx.NewSubTest("reconnect reuses WorkloadEntry").Run(func(ctx framework.TestContext) {
 				// ensure we have two pilot instances, other tests can pass before the second one comes up
@@ -114,7 +114,7 @@ func TestVMRegistrationLifecycle(t *testing.T) {
 						return errors.New("expected 2 pilots")
 					}
 					return nil
-				}, retry.Timeout(10*time.Second))
+				}, retry.Timeout(5*time.Minute))
 
 				// get the initial workload entry state
 				entries := getWorkloadEntriesOrFail(ctx, autoVM)
@@ -138,7 +138,7 @@ func TestVMRegistrationLifecycle(t *testing.T) {
 						return errors.New("expected WorkloadEntry to be updated by other pilot")
 					}
 					return nil
-				}, retry.Delay(5*time.Second))
+				}, retry.Delay(5*time.Minute))
 			})
 			ctx.NewSubTest("disconnect deletes WorkloadEntry").Run(func(ctx framework.TestContext) {
 				deployment := fmt.Sprintf("%s-%s", autoVM.Config().Service, "v1")
