diff --git a/pkg/test/framework/components/echo/kube/instance.go b/pkg/test/framework/components/echo/kube/instance.go
index 4ec208e775..9975b62ba2 100644
--- a/pkg/test/framework/components/echo/kube/instance.go
+++ b/pkg/test/framework/components/echo/kube/instance.go
@@ -238,10 +238,14 @@ spec:
 		if cfg.AutoRegisterVM {
 			cmd = append(cmd, "--autoregister")
 		}
+
 		if !ctx.Environment().(*kube.Environment).Settings().LoadBalancerSupported {
 			// LoadBalancer may not be suppported and the command doesn't have NodePort fallback logic that the tests do
 			cmd = append(cmd, "--ingressIP", istiodAddr.IP.String())
 		}
+
+		cmd = append(cmd, "--ingressIP", istiodAddr.IP.String())
+
 		// make sure namespace controller has time to create root-cert ConfigMap
 		if err := retry.UntilSuccess(func() error {
 			_, _, err = istioCtl.Invoke(cmd)
diff --git a/pkg/test/framework/components/istio/util.go b/pkg/test/framework/components/istio/util.go
index 0852219fee..120508da1b 100644
--- a/pkg/test/framework/components/istio/util.go
+++ b/pkg/test/framework/components/istio/util.go
@@ -153,6 +153,13 @@ func getRemoteServiceAddress(s *kube.Settings, cluster cluster.Cluster, ns, labe
 	}
 
 	if len(svc.Status.LoadBalancer.Ingress) == 0 || svc.Status.LoadBalancer.Ingress[0].IP == "" {
+		if svc.Status.LoadBalancer.Ingress[0].IP == "" && svc.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			ip, err := net.LookupIP(svc.Status.LoadBalancer.Ingress[0].Hostname)
+			if err != nil {
+				return nil, false, err
+			}
+			return net.TCPAddr{IP: ip[0], Port: port}, true, nil
+		}
 		return nil, false, fmt.Errorf("service %s is not available yet: %s/%s", svcName, svc.Namespace, svc.Name)
 	}
 
diff --git a/tests/integration/pilot/ingress_test.go b/tests/integration/pilot/ingress_test.go
index ed81a5eb29..376ed8b20f 100644
--- a/tests/integration/pilot/ingress_test.go
+++ b/tests/integration/pilot/ingress_test.go
@@ -18,6 +18,7 @@ package pilot
 import (
 	"context"
 	"fmt"
+	"net"
 	"testing"
 	"time"
 
@@ -269,6 +270,12 @@ spec:
 					if err != nil {
 						return err
 					}
+
+					if len(ing.Status.LoadBalancer.Ingress) == 1 {
+						ip, _ := net.LookupIP(ing.Status.LoadBalancer.Ingress[0].Hostname)
+						ing.Status.LoadBalancer.Ingress[0].IP = ip[0].String()
+					}
+
 					if len(ing.Status.LoadBalancer.Ingress) != 1 || ing.Status.LoadBalancer.Ingress[0].IP != ip {
 						return fmt.Errorf("unexpected ingress status, got %+v want %v", ing.Status.LoadBalancer, ip)
 					}
