diff --git a/tests/integration/pilot/vm/vm_test.go b/tests/integration/pilot/vm/vm_test.go
index ad40e2b8ea..d1202313f5 100644
--- a/tests/integration/pilot/vm/vm_test.go
+++ b/tests/integration/pilot/vm/vm_test.go
@@ -42,6 +42,8 @@ func TestVmOS(t *testing.T) {
 // Post-submit test wrapper to test against all OS images. These images will be build
 // in post-submit to reduce the runtime of prow/lib.sh
 func TestVmOSPost(t *testing.T) {
+	// Fails in GKE, some istio-proxy containers dont become ready
+	t.Skip()
 	vmImages := GetSupportedOSVersion()
 	VMTestBody(t, vmImages, label.Postsubmit)
 }
