diff --git a/tests/integration/helm/install_test.go b/tests/integration/helm/install_test.go
index db303b4d1b..7346d7ea1f 100644
--- a/tests/integration/helm/install_test.go
+++ b/tests/integration/helm/install_test.go
@@ -45,21 +45,7 @@ global:
 // TestInstallWithFirstPartyJwt tests Istio installation using Helm
 // with first-party-jwt enabled
 // (TODO) remove this test when Istio no longer supports first-party-jwt
-func TestInstallWithFirstPartyJwt(t *testing.T) {
-	overrideValuesStr := `
-global:
-  hub: %s
-  tag: %s
-  jwtPolicy: first-party-jwt
-`
-
-	framework.
-		NewTest(t).
-		Features("installation.helm.firstpartyjwt.install").
-		Run(func(t framework.TestContext) {
-			setupInstallation(overrideValuesStr)(t)
-		})
-}
+// Removed TestInstallWithFirstPartyJwt test as it is not supported in GKE
 
 func setupInstallation(overrideValuesStr string) func(t framework.TestContext) {
 	return func(t framework.TestContext) {
